pkg/services/datasources/models.go

```diff
@@ -77,6 +79,18 @@ type TeamHTTPHeader struct {
 	Value  string `json:"value"`
 }
 
+// TeamHTTPHeaderValueRegexMatch returns a regex that can be used to check
+// words separated by special characters
+// "namespace!=auth", "env="prod", "env!~dev"
+func (th *TeamHTTPHeader) TeamHTTPHeaderValueRegexMatch() (*regexp.Regexp, error) {
+	exp := `^[\w\-\.\*\!\~\=]+$`
+	return regexp.Compile(exp)
+}
+
+func (th TeamHTTPHeaders) ValidHeaders() []string {
```

Comment:
The same as for the previous function. Seems, it's better to move it to the datasource.go since all validation happens there.

Reply:
but it's valid headers only for the teamHTTPHeaders not the datasource itself

Reply:
This `models.go` is also shared code for all data sources, so there's no difference. I mean this function does nothing related to the data model, it's a part of validation.

Reply:
moved to datasources.go close to the validation function